crates/swc_ecma_minifier/src/compress/pure/misc.rs

```diff
@@ -247,76 +248,78 @@ impl Pure<'_> {
 
     /// `new RegExp("([Sap]+)", "ig")` => `/([Sap]+)/gi`
     fn optimize_regex(&mut self, args: &mut Vec<ExprOrSpread>, span: &mut Span) -> Option<Expr> {
-        if args.is_empty() || args.len() > 2 {
-            return None;
+        fn valid_pattern(pattern: &Expr) -> Option<JsWord> {
+            if let Expr::Lit(Lit::Str(s)) = pattern {
+                if s.value.contains(|c: char| {
+                    // whitelist
+                    !c.is_ascii_alphanumeric()
+                        && !matches!(c, '$' | '[' | ']' | '(' | ')' | '{' | '}' | '-' | '+' | '_')
+                }) {
+                    return None;
+                }
+                if s.value.contains("\\\0") || s.value.contains('/') {
```

Comment:
Nit: I think this is redundant? I think both the `\0` null byte and `/` will be caught be the above `!is_ascii_alphanumeric() && !matches()`?